crates/module-system/module-implementations/sov-evm/tests/integration/transactions.rs

```diff
@@ -131,3 +131,69 @@ fn test_account_nonce() {
         }),
     });
 }
+
+// Check that if the same account deploys two contracts, each deployment results in a unique contract address
+#[test]
+fn test_deploy_many_contracts() {
+    let (mut runner, account, _) = setup();
+    let contract = SimpleStorageContract::default();
+    let contract_addr_1 = account.address().create(0);
+
+    // Deploy a contract and execute two transactions that update its storage.
+    {
+        let create_contract_tx_1 = create_deploy_tx(0, &contract, &account);
+        let set_value_tx_1 = create_set_arg_tx(5, 1, &contract, contract_addr_1, &account);
+        let set_value_tx_12 = create_set_arg_tx(7, 2, &contract, contract_addr_1, &account);
+
+        let evm = Evm::<S>::default();
+        runner.execute_batch(BatchTestCase {
+            input: vec![create_contract_tx_1, set_value_tx_1, set_value_tx_12].into(),
+            assert: Box::new(move |_ctx, state| {
+                let storage_value_1 = evm
+                    .get_storage(&contract_addr_1, &U256::ZERO, state)
+                    .unwrap()
+                    .unwrap();
+
+                assert_eq!(U256::from(7), storage_value_1);
+            }),
+        });
+    }
+
+    // Deploy another contract and execut single transaction that updates its storage.
+    {
+        let contract_addr_2 = account.address().create(3);
+        let create_contract_tx_2 = create_deploy_tx(3, &contract, &account);
+        let set_value_tx_2 = create_set_arg_tx(99, 4, &contract, contract_addr_2, &account);
+
+        let evm = Evm::<S>::default();
+        runner.execute_batch(BatchTestCase {
+            input: vec![create_contract_tx_2, set_value_tx_2].into(),
+            assert: Box::new(move |_ctx, state| {
+                // The two contracts have different addresses.
+                assert_ne!(contract_addr_1, contract_addr_2);
```

Comment:
This assertion doesn't test our code but tests the library.
It would make sense if we assert actual addresses of created contracts, but not statically precomputed addresses

Reply:
If we were testing opaque contracts that can use proxy patterns - sure. But here we test our dummy contract and we know it doesn't do anything like that.
In this test we want to check that we do infact increment nonce on contract deployment and we can check that by just checking the resulting addresses. That way if we ever break that behavior - we have a small test that tests just that and it's easier to locate and debug and reason about